file_paths.md

```diff
@@ -106,16 +115,144 @@ puppet-agent
+        hiera.yaml                            # :hiera_config
+        hieradata                             # n/a
+        modules                               # user modulepath
+
+    C:\ProgramData\PuppetLabs\mcollective\etc                                same
+        client.cfg
+        facts.yaml
+        server.cfg
+
+    C:\ProgramData\PuppetLabs\mcollective\var                                same
+        log
+            mcollective.log
+
+    C:\ProgramData\PuppetLabs\puppet\etc      # :confdir                     same
+        csr_attributes.yaml                   # :csr_attributes
+        custom_trusted_oid_mapping.yaml       # :trusted_oid_mapping_file
+        puppet.conf                           # :config
+        ssl                                   # :ssldir
+
+    C:\ProgramData\PuppetLabs\puppet\cache    # :vardir                      C:\ProgramData\PuppetLabs\puppet\var
+        bucket                                # :bucketdir
+        client_yaml                           # :clientyamldir
+        client_data                           # :client_datadir
+        clientbucket                          # :clientbucketdir
+        devicedir                             # :devices
+        facts.d                               # :pluginfactdest (pluginsync'ed)
+        lib                                   # :libdir
+        facts                                 # used to generate :factpath
+        puppet-module                         # :module_working_dir
+        reports                               # :reportdir
+        server_datadir                        # :server_data
+        state                                 # :statedir
+        yaml                                  # :yamldir
+
+    C:\ProgramData\PuppetLabs\facter                                         same
+        facts.d                               # external facts directory (not pluginsync'ed)
+
+    C:\ProgramData\PuppetLabs\mcollective
+        plugins                               # user installed plugins
+
+    C:\ProgramData\PuppetLabs\puppet\var\log  # :logdir                      same
+        puppet.log                            # not enabled by default
+
+    C:\ProgramData\PuppetLabs\puppet\var\run  # :rundir                      same
+        agent.pid                             # :pidfile
+
+    C:\Program Files\Puppet Labs\Puppet\bin
+        cfacter.bat                           # bat file wrappers
+        facter.bat
+        hiera.bat
+        mco.bat
+        puppet.bat
+        environment.bat                       # setup LOAD_PATH
+        *_interactive.bat                     # targets for shortcuts
+
+    C:\Program Files\Puppet Labs\Puppet\cfacter
+        bin                                   # executables and dlls
+            cfacter.exe
+            libfacter.so
+            lib*.dll
+        inc                                   # cfacter headers
+            facter
```

Comment:
TBH, I'm not super happy with the one-bin/lib per project approach, as for each new project, e.g. foo, we have to add `foo/bin` to `PATH` and `foo/lib` to `LOAD_PATH` in
https://github.com/puppetlabs/puppet_for_the_win/blob/master/conf/windows/stage/bin/environment.bat#L25.

We then have to add a `.bat` wrapper in `C:\Program Files\Puppet Labs\Puppet\bin\foo.bat` to call the ruby bin wrapper `C:\Program Files\Puppet Labs\Puppet\foo\bin\foo`.

That said, I think changing this would be a fair amount of work, probably not worth the trouble.